CLI/src/Daemon.cs

```diff
@@ -50,188 +43,194 @@ class Config {
+            try
+            {
                 List<string> args;
-		        try {
+                try
+                {
 
                     args = p.Parse(argv);
-                } catch (OptionException ex) {
+                }
+                catch (OptionException ex)
+                {
                     return exit(1, ex.Message);
-		        }
-
-		        if (help) {
-		            p.WriteOptionDescriptions(Console.Out);
-		            return 0;
-		        }
-		        if (version) {
-		            Console.WriteLine(PROGVERSION);
-		            return 0;
-		        }
-		        if (config.OutputFiles.Count > 0 && config.InputFiles.Count != config.OutputFiles.Count)
-		            return exit(2, "The number of output files must be equal to the number of input files.");
-		        if (config.OutputFiles.Count == 0 && config.Codegen)
-		            foreach(var path in config.InputFiles) config.OutputFiles.Add(path+".cee");
-
-		        if (args.Count > 0) pipename = args[0];
-
-		        if (once) {
-		            runOnce(config);
-		        } else {
-		            runServer(pipename);
-		        }
-		    } catch (Exception e) {
+                }
+
+                if (help)
+                {
+                    p.WriteOptionDescriptions(Console.Out);
+                    return 0;
+                }
+                if (version)
+                {
+                    Console.WriteLine(PROGVERSION);
+                    return 0;
+                }
+                if (config.OutputFiles.Count > 0 && config.InputFiles.Count != config.OutputFiles.Count)
+                    return exit(2, "The number of output files must be equal to the number of input files.");
+                if (config.OutputFiles.Count == 0 && config.Codegen)
+                    foreach (var path in config.InputFiles) config.OutputFiles.Add(path + ".cee");
+
+                if (args.Count > 0) pipename = args[0];
+
+                //"startClient" will be true when "-K" is passed as an argument in command line.
+                if (startClient && once)
+                {
+#if DEBUG
+                    var watch = System.Diagnostics.Stopwatch.StartNew();
+#endif
+                    using (NamedPipeClientStream namedPipeClient = new NamedPipeClientStream(pipename))
+                    {
+                        try
+                        {
+                            namedPipeClient.Connect(100);
+                        }
+                        catch (TimeoutException tEx)
+                        {
+                            System.Diagnostics.Process process = new System.Diagnostics.Process();
+                            System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+                            startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                            startInfo.FileName = "cmd.exe";
+                            startInfo.Arguments = @"/c TypeCobol.CLI.exe";
```

Comment:
One thing I didn't thought about is that TypeCobol.CLI is not necessarily in the path.
So I see 2 possible solutions:
- Can you retrieve the current folder of the TypeCobol.CLI.exe and then use it to launch this command
- If it doesn't works, then the path to TypeCobol.CLI.exe need to be passed as an argument (with option -K)

Do you see another solution?